src/Diagnose/PHPStanDiagnoseExtension.php

```diff
@@ -48,11 +53,36 @@ public function print(Output $output): void
 			'<info>PHP runtime version:</info> %s',
 			$phpRuntimeVersion->getVersionString(),
 		));
-		$output->writeLineFormatted(sprintf(
-			'<info>PHP version for analysis:</info> %s (from %s)',
-			$this->phpVersion->getVersionString(),
-			$this->phpVersion->getSourceLabel(),
-		));
+
+		$composerPhpVersion = $this->getComposerRequireVersion();
+		if ($composerPhpVersion !== null) {
+			$output->writeLineFormatted(sprintf(
+				'<info>PHP composer.json required version:</info> %s',
+				$composerPhpVersion,
+			));
+		}
+
+		if (
+			$this->phpVersion->getSource() === PhpVersion::SOURCE_CONFIG
+			&& is_array($this->configPhpVersion)
+		) {
+			$minVersion = new PhpVersion($this->configPhpVersion['min']);
+			$maxVersion = new PhpVersion($this->configPhpVersion['max']);
+
+			$output->writeLineFormatted(sprintf(
+				'<info>PHP version for analysis:</info> %s-%s (from %s)',
+				$minVersion->getVersionString(),
+				$maxVersion->getVersionString(),
+				$this->phpVersion->getSourceLabel(),
+			));
+
+		} else {
+			$output->writeLineFormatted(sprintf(
+				'<info>PHP version for analysis:</info> %s (from %s)',
```

Comment:
What is this, isn't phpstan using only one version for the analysis?

Reply:
Since https://github.com/phpstan/phpstan-src/pull/3584 you can declare a range